instrumentedProfilingLibrary/Time Profiler/ProfileDataManager.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ProfileDataManager.h"
+#include <iostream>
+
+using namespace std;
+
+ProfileDataManager* ProfileDataManager::profiledatamanager = nullptr;
+
+ProfileDataManager::ProfileDataManager(const string name){
+	
+	myfile.open("time.txt");
+	
+	if (myfile.is_open()) {
+	    cout << "myfile is open \n" << endl;
+	}
+
+}
+
+
+ProfileDataManager:: ~ProfileDataManager() {
+	
+	myfile.close();
+
+}
+
+
+ProfileDataManager* ProfileDataManager::GetInstance(const string& value) {

```

Comment:
yet another design issue: say you call `GetInstance("file_one")` and then call it with `GetInstance("file_two")`.
What does the user expect to happen?
Open two different files?
Close the existing one, open a new one?
Ignore the second file name if it exists?
None of these options seems quite optimal. I'd say it's probably easier to give the user access to a lightweight instance that doesn't also perform the file manangement effort when the instance is created.

Reply:
I fixed this and made the instance to be 'getInstance()', I had not thought  of it from this angle.